client/src/pages/thread/thread.jsx

```diff
@@ -76,21 +51,22 @@ const Thread = () => {
   const handleToggleShowOwnPosts = useCallback(() => {
     const currentUserId = showOwnPosts ? userId : undefined;
 
-    handlePostsLoad({ ...postsFilterInitialState, userId: currentUserId });
-
     dispatchPostsFilter({
       type: PostsFilterAction.TOGGLE_SHOW_OWN_POSTS,
       payload: {
-        userId: currentUserId,
-        from: postsFilter.count
+        userId: currentUserId
       }
     });
-  }, [showOwnPosts, userId, handlePostsLoad]);
+  }, [showOwnPosts, userId, dispatchPostsFilter]);
 
   useEffect(() => {
     handleToggleShowOwnPosts();
   }, [showOwnPosts, handleToggleShowOwnPosts]);
 
+  useEffect(() => {
+    handlePostsLoad({ from: 0, count, userId: postsFilter.userId });
```

Comment:
This component won't know anything about from and count, use it only in redux actions